packages/aws-cdk-lib/core/test/app.test.ts

```diff
@@ -78,8 +78,8 @@ describe('app', () => {
       '/stack1/s1c1': [{ type: 'aws:cdk:logicalId', data: 's1c1' }],
       '/stack1/s1c2':
         [{ type: 'aws:cdk:logicalId', data: 's1c2' },
-          { type: 'aws:cdk:warning', data: 'warning1' },
-          { type: 'aws:cdk:warning', data: 'warning2' }],
+          { type: 'aws:cdk:warning', data: { id: 'warning1', message: 'warning1', scope: 'stack1/s1c2' } },
```

Comment:
This changes the type associated with `aws:cdk:warning`. I don't think that's safe if we consider external processors.

Either we now have a new type `aws:cdk:warningv2`, or we find a way to add the new information additively into the old schema. 

But if we can keep warning filtering entirely in the CDK app, there's no need to change this format at all. We can turn the message of a `warningv2` into:

```
{ type: 'aws:cdk:warning', data: '[warning1] warning message here' }
```

And it'll be backwards compatible.

Reply:
Or honestly, a better message would probably be:

```
warning message here [warning1]
```